packages/flutter_tools/test/integration.shard/vmservice_integration_test.dart

```diff
@@ -28,22 +26,21 @@ void main() {
 
       flutter = FlutterRunTestDriver(tempDir);
       await flutter.run(withDebugger: true);
-      final int port = flutter.vmServicePort;
+      final int? port = flutter.vmServicePort;
       vmService = await vmServiceConnectUri('ws://localhost:$port/ws');
     });
 
     tearDown(() async {
-      await flutter?.stop();
+      await flutter.stop();
       tryToDelete(tempDir);
     });
 
     testWithoutContext('getSupportedProtocols includes DDS', () async {
-      final ProtocolList protocolList =
-          await vmService.getSupportedProtocols();
+      void expectDDS(Protocol protocol) =>
+          expect(protocol.protocolName, anyOf('VM Service', 'DDS'));
+      final ProtocolList protocolList = await vmService.getSupportedProtocols();
       expect(protocolList.protocols, hasLength(2));
-      for (final Protocol protocol in protocolList.protocols) {
-        expect(protocol.protocolName, anyOf('VM Service', 'DDS'));
-      }
+      protocolList.protocols?.forEach(expectDDS);
```

Comment:
I think this would be easier to read if you leave the `for` loop and just add `!` since this is just a test and we don't need to worry about user-facing crashes.